test/5.future-and.test.js

```diff
@@ -117,11 +69,51 @@ describe('FutureAnd', () => {
   describe('#toString()', () => {
 
     it('returns the code to create the FutureAnd', () => {
-      const m = Future.of(1).and(Future.of(2));
+      const m = and(Future.of(1), Future.of(2));
       const s = 'Future.of(1).and(Future.of(2))';
       expect(m.toString()).to.equal(s);
     });
 
   });
 
+};
+
+describe('Future.and()', () => {
+
+  it('is a curried binary function', () => {
+    expect(Future.and).to.be.a('function');
+    expect(Future.and.length).to.equal(2);
+    expect(Future.and(F.resolved)).to.be.a('function');
+  });
+
+  it('throws when not given a Future as first argument', () => {
+    const f = () => Future.and(1);
+    expect(f).to.throw(TypeError, /Future.*first/);
+  });
+
+  it('throws when not given a Future as second argument', () => {
+    const f = () => Future.and(Future.of(1), 1);
+    expect(f).to.throw(TypeError, /Future.*second/);
+  });
+
+  //TODO: Argument order.
+  testInstance((a, b) => Future.and(a, b));
```

Comment:
`Future.and` was one of the functions I implemented without a `createDispatcher` from the start, and I made the same mistake there that caused all this mess now. Essentially the argument order of `Future.and` is the wrong way around (static functions should be flipped). If I'd fix that now, it would be a breaking change, hence the `#TODO`.